scripts/changelog.js

```diff
@@ -1,434 +0,0 @@
-  const date = args.endTag ? exec(`git log -1 --date=short --format=%ad ${args.endTag}`) : today()
-
-  const output = logger(RELEASE.title(version, date) + '\n')
-
-  for (const key of Object.keys(commits)) {
-    if (commits[key].length > 0) {
-      output.group(`### ${key}\n`)
-
-      for (const commit of commits[key]) {
-        let groupCommit = `* [\`${commit.hash}\`](${commit.url})`
-        for (const pr of commit.prs) {
-          groupCommit += ` [#${pr.number}](${pr.url})`
-        }
-        groupCommit += ` ${commit.title}`
-        if (key !== 'Dependencies') {
-          for (const user of commit.credit) {
-            if (args.type === 'gh') {
-              groupCommit += ` (${user.name})`
-            } else {
-              groupCommit += ` ([${user.name}](${user.url}))`
-            }
-          }
-        }
-
-        output.group(groupCommit)
-        // only optionally add full commit bodies to changelog
-        if (commit.body && commit.body.length && args.format === 'long') {
-          output.log(commit.body)
-        }
-        output.groupEnd()
-      }
-
-      output.log()
-      output.groupEnd()
-    }
-  }
-
-  return {
-    date,
-    version,
-    release: output.toString(),
-  }
-}
-
-const main = async (argv) => {
-  const args = parseArgs(argv)
-
-  if (args.read) {
-    // this reads the release notes for that version
-    let { release } = findRelease(args, args.tag)
-    if (args.type === 'gh') {
-      // changelog was written in markdown so convert user links to gh release style
-      // XXX: this needs to be changed if the `generateRelease` format changes
-      release = release.replace(/\(\[(@[a-z\d-]+)\]\(https:\/\/github.com\/[a-z\d-]+\)\)/gi, '($1)')
-    }
-    return console.log(release)
-  }
-
-  // otherwise fetch the requested release from github
-  const { release, version, date } = await generateRelease(args)
-
-  if (args.write) {
-    // only try and run release manager issue update on write since that signals
-    // the first time we know the version of the release
-    try {
-      exec(
-        `node scripts/release-manager.js --update --version=${version.slice(1)} --date=${date}`, {
-          stdio: 'pipe',
-        })
-    } catch (e) {
-      console.error(`Updating release manager issue failed: ${e.stderr}`)
-    }
-
-    const { release: existing, changelog } = findRelease(args, version)
-    fs.writeFileSync(
-      args.file,
-      existing
-        // replace existing release with the newly generated one
-        ? changelog.replace(existing, release)
-        // otherwise prepend a new release at the start of the changelog
-        : changelog.replace(RELEASE.h1, RELEASE.h1 + release + RELEASE.sep),
-      'utf-8'
-    )
-    return console.log(
-      `Release notes for ${version} written to "./${relative(process.cwd(), args.file)}".`
-    )
-  }
-
-  console.log(release)
-}
-
-main(process.argv.slice(2))
```

Comment:
The relevant parts of this file have been moved to `template-oss`. The cool part is we now use our custom changelog in every package.